ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/mapper/defendant/ResponseMapper.java

```diff
@@ -0,0 +1,274 @@
+                .map(evidenceRowMapper::to)
+                .filter(Objects::nonNull)
+                .map(row -> CCDCollectionElement.<CCDEvidenceRow>builder().value(row).build())
+                .collect(Collectors.toList())
+            );
+        };
+    }
+
+    private Consumer<PaymentDeclaration> mapPaymentDeclaration(CCDDefendant.CCDDefendantBuilder builder) {
+        return paymentDeclaration -> {
+            builder.paymentDeclarationExplanation(paymentDeclaration.getExplanation());
+            builder.paymentDeclarationPaidDate(paymentDeclaration.getPaidDate());
+        };
+    }
+
+    public void from(Claim.ClaimBuilder claimBuilder, CCDDefendant defendant) {
+        requireNonNull(claimBuilder, "claimBuilder must not be null");
+        requireNonNull(defendant, "defendant must not be null");
+        if (defendant.getResponseType() == null) {
+            return;
+        }
+
+        switch (defendant.getResponseType()) {
+            case FULL_DEFENCE:
+                claimBuilder.response(extractFullDefence(defendant));
+                break;
+            case FULL_ADMISSION:
+                claimBuilder.response(extractFullAdmission(defendant));
+                break;
+            case PART_ADMISSION:
+                claimBuilder.response(extractPartAdmission(defendant));
+                break;
+            default:
+                throw new MappingException("Invalid responseType");
+        }
+    }
+
+    private FullDefenceResponse extractFullDefence(CCDDefendant defendant) {
+        return FullDefenceResponse.builder()
+            .defendant(defendantPartyMapper.from(defendant))
+            .statementOfTruth(extractStatementOfTruth(defendant))
+            .moreTimeNeeded(YesNoOption.valueOf(defendant.getResponseMoreTimeNeededOption().name()))
+            .freeMediation(YesNoOption.valueOf(defendant.getResponseMoreTimeNeededOption().name()))
+            .defenceType(DefenceType.valueOf(defendant.getResponseDefenceType().name()))
+            .defence(defendant.getResponseDefence())
+            .evidence(extractDefendantEvidence(defendant))
+            .timeline(extractDefendantTimeline(defendant))
+            .paymentDeclaration(extractPaymentDeclaration(defendant))
+            .build();
+    }
+
+    private StatementOfTruth extractStatementOfTruth(CCDDefendant defendant) {
+        String signerName = defendant.getResponseDefendantSOTSignerName();
+        String signerRole = defendant.getResponseDefendantSOTSignerRole();
+        if (isAllBlank(signerName, signerRole)) {
+            return null;
+        } else {
+            return new StatementOfTruth(signerName, signerRole);
+        }
+    }
+
+    private PaymentDeclaration extractPaymentDeclaration(CCDDefendant defendant) {
+        LocalDate paidDate = defendant.getPaymentDeclarationPaidDate();
+        String explanation = defendant.getPaymentDeclarationExplanation();
+        if (paidDate == null && explanation == null) {
```

Comment:
you can use isAllBlank here too

Reply:
Can't use here, as `isAllBlank` needs all `String` params